File: textconvert.py
import time

def convert_to_upper(input_file, output_file):
    with open(input_file, 'r') as input_f, open(output_file, 'w') as output_f:
        for line in input_f:
            output_f.write(line.upper())

def measure_time(input_file):
    output_file = 'output.txt'
    start_time = time.time()

    convert_to_upper(input_file, output_file)

    end_time = time.time()
    time_taken = end_time - start_time

    return time_taken

if __name__ == '__main__':
    file_sizes = ['200MB.txt', '400MB.txt', '600MB.txt', '800MB.txt', '1000MB.txt']
    time_taken = []

    for file in file_sizes:
        time_taken.append(measure_time(file))

    # Print the time taken
    for i in range(len(file_sizes)):
        print(f"Time taken for {file_sizes[i]}: {time_taken[i]} seconds")

File: textconvert.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

public class ConvertToUpper {
    public static void convertToUpper(String inputFile, String outputFile) throws IOException {
        BufferedReader reader = new BufferedReader(new FileReader(inputFile));
        BufferedWriter writer = new BufferedWriter(new FileWriter(outputFile));
        String line;
        
        while ((line = reader.readLine()) != null) {
            writer.write(line.toUpperCase());
            writer.newLine();
        }
        
        reader.close();
        writer.close();
    }
    
    public static double measureTime(String inputFile) throws IOException {
        String outputFile = "output.txt";
        long startTime = System.nanoTime();
        
        convertToUpper(inputFile, outputFile);
        
        long endTime = System.nanoTime();
        double timeTaken = (endTime - startTime) / 1e9;
        
        return timeTaken;
    }
    
    public static void main(String[] args) throws IOException {
        String[] fileSizes = {"200MB.txt", "400MB.txt", "600MB.txt", "800MB.txt", "1000MB.txt"};
        double[] timeTaken = new double[5];
        
        for (int i = 0; i < 5; i++) {
            timeTaken[i] = measureTime(fileSizes[i]);
        }
        
        // Print the time taken
        for (int i = 0; i < 5; i++) {
            System.out.printf("Time taken for %s: %.6f seconds\n", fileSizes[i], timeTaken[i]);
        }
    }
}
